src/communication/context.cpp

```diff
@@ -78,7 +78,10 @@ bool ClientContext::use_ssl() { return use_ssl_; }
 
 ServerContext::ServerContext(const std::string &key_file, const std::string &cert_file, const std::string &ca_file,
                              bool verify_peer) {
-  ctx_.emplace(boost::asio::ssl::context::tls_server);
+  ctx_.emplace(boost::asio::ssl::context::sslv23_server);
+  // NOLINTNEXTLINE(hicpp-signed-bitwise)
+  ctx_->set_options(boost::asio::ssl::context::default_workarounds | boost::asio::ssl::context::no_sslv2 |
+                    boost::asio::ssl::context::single_dh_use);
```

Comment:
If you check their definition, they are removed in OpenSSL 1.1.0. Don't we use only OpenSSL 1.1.0 and later versions? If so, then this is not necessary.

Reply:
I think we want to support OpenSSL 1.0.0 and 1.1.0 since we also support native OpenSSL on Centos 7.

Reply:
Khm... CentOS... :smile: Sorry then, you are right.